### PR TITLE
dev/core#670 - Fix Saving of Case Activity tags

### PR DESCRIPTION
**Pull request by deb1990**
Overview
----------------------------------------
When editing an Activity for cases, the selected `tags` are not saved.
This PR fixes the bug.

Before
----------------------------------------
before

After
----------------------------------------
after

Technical Details
----------------------------------------
In https://github.com/civicrm/civicrm-core/blob/3bf2661bd263d718e867ad5170965d1bcb110df6/CRM/Case/Form/Activity.php#L546 it is assumed that `$params['tag']` is an array, but actually is a comma separated string.

But in the parent class `CRM_Activity_Form_Activity`, this feature works because of the code https://github.com/civicrm/civicrm-core/blob/43076c76f4724d0f42ceb03b46acfbce2c751233/CRM/Activity/Form/Activity.php#L1066, which converts the string to an Array.

So similar code needs to be implemented in `CRM_Case_Form_Activity`. 
```php
if (!empty($params['tag'])) {
  if (!is_array($params['tag'])) {
    $params['tag'] = explode(',', $params['tag']);
  }
  foreach ($params['tag'] as $tag) {
    $tagParams[$tag] = 1;
  }
}
```

Also the following code has been refactored to use  `array_fill_keys`.
```
foreach ($params['tag'] as $tag) {
  $tagParams[$tag] = 1;
}
```

### CODE REVIEW

File: CRM/Case/Form/Activity.php [hunk 1]
```diff
@@ -543,9 +543,10 @@ public function postProcess($params = NULL) {
         // add tags if exists
         $tagParams = array();
         if (!empty($params['tag'])) {
-          foreach ($params['tag'] as $tag) {
-            $tagParams[$tag] = 1;
+          if (!is_array($params['tag'])) {
+            $params['tag'] = explode(',', $params['tag']);
           }
+          $tagParams = array_fill_keys($params['tag'], 1);
```
Comment (swastikpareek) [inline CRM/Case/Form/Activity.php]: Add a new line before ending if statement

**Review by swastikpareek — COMMENTED** [review]
(*CiviCRM Review Template DEL-1.2*)


* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained with a link (JIRA, Github, Gitlab, StackExchange).
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
  
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: There are relevant updates for the documentation.
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __UNREVIEWED__
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
        * __ISSUE__: We need to reformat the code to match the standard

File: CRM/Case/Form/Activity.php (hunk 1 above)
Comment (deb1990) [reply CRM/Case/Form/Activity.php]: Fixed

**Review by swastikpareek — APPROVED** [review]
(*CiviCRM Review Template DEL-1.2*)


* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained with a link (JIRA, Github, Gitlab, StackExchange).
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
  
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: There are relevant updates for the documentation.
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __UNREVIEWED__
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
        * __PASS__: The functionality, purpose, and style of the code seems clear+sensible.

**Comment by colemanw** [comment]
@civicrm-builder retest this please

**Comment by colemanw** [comment]
@deb1990 could you please squash these three commits to one and then force-push?
As a general rule we like to have a 1-to-1 ratio of PRs and commits (except for very big PRs which can be logically broken into multiple commits). This keeps our main commit history clean to avoid things like:

> Commit 1: Fix X
> Commit 2: Better fix for X
> Commit 3: Fix typo in previous commit

**Comment by deb1990** [comment]
@colemanw Commit history fixed. please check again.

**Pull request merged**